### PR TITLE
Feature: basic skeleton + basic routing

### PR DESCRIPTION
**Pull request by vadymmarkov**
@zenangst @RamonGilabert @onmyway133 
1) Add basic `Request` and `Response`.
2) Add `ResourceController` protocol for convenient mapping to routes.
3) Add basic router.

``` swift
let application = Application()
let controller = CatController()

application.resources("cats", controller)
// GET /cats  => controller.index
// GET /cats/new  => controller.new
// GET /cats/{id}  => controller.show
// GET) /cats/{id}/edit => controller.edit
// POST /cats => controller.create
// DELETE /cats/{id} => controller.destroy
// PATCH /cats/{id} => controller.update
```

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Licensed under the **MIT** license
 > included in all copies or substantial portions of the Software.
 >
 > THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-> EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+> EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THkE WARRANTIES OF
```
Comment (zenangst) [inline LICENSE.md]: 🤔

File: Tests/Flamingo/Router/RouterSpec.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        it("adds all resource routes") {
+          expect(router.routes.filter({ self.isRoute($1, "/cats", .GET) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats/new", .GET) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats/{id}", .GET) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats/{id}/edit", .GET) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats", .POST) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats/{id}", .DELETE) }).count).to(equal(1))
+          expect(router.routes.filter({ self.isRoute($1, "/cats/{id}", .PATCH) }).count).to(equal(1))
```
Comment (zenangst) [inline Tests/Flamingo/Router/RouterSpec.swift:28]: 😹

File: LICENSE.md (hunk 1 above)
Comment (vadymmarkov) [inline LICENSE.md]: @zenangst Have no idea where it came from :smile: Fixed.

**Comment by onmyway133** [comment]
:rocket: 
`carthage bootstrap --platform OSX` and `Cmd+U` works for me

**Pull request merged**